go/src/url_shortener/server_test.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestAddURL(t *testing.T) {
+	sut := URLShortener{
+		port: 9090,
+
+		expanderRoute:   "/",
+		shortenRoute:    "/shorten/",
+		statisticsRoute: "/statistics",
+
+		mappings: make(map[string]string),
+
+		statistics: NewStatsJSON(),
+	}
```

Comment:
It seems the URLShortener struct creation is the same for multiple tests so it could be useful create a function to make this a one-liner.
FIY: nothing in a `_test.go` file gets compiled in the final binary so a function such as the one I'm talking about would not increase the binary size.